### PR TITLE
Prepare for building Topology2.0 topologies

### PR DESCRIPTION
**Pull request by ranj063**
In preparation for Topology2.0, move the current topology files
to the topology1 folder and once the 1.0 topologies are
built copy them to the /sof/tools/build-tools/topology folder.

When Topology2.0 topologies come along, they will be built into
the topology2 folder and the 2.0 binaries will be copied over
the 1.0 binaries.

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]
I guess the build scripts wont need updated as changes will be incorporated in the makefiles ?

**Comment by lgirdwood** [comment]
@ranj063 I've finished merging for tonight - good time to fix the conflicts now :)

**Comment by ranj063** [comment]
> @ranj063 I've finished merging for tonight - good time to fix the conflicts now :)

@lgirdwood, i've fixed the conflicts now but there are a bunch of topology PR's pending. Should we merge those before merging this?

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+# verbose or not.
+	add_custom_command(
+		OUTPUT ${output}.tplg
+		COMMAND alsatplg \$\${VERBOSE:+-v 1} -c ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf -o ${output}.tplg
+		MAIN_DEPENDENCY ${output}.conf
+		USES_TERMINAL
+	)
+
+	add_custom_command(
+		OUTPUT final_${output}.tplg
```
Comment (marc-hb) [inline tools/topology/topology1/CMakeLists.txt:208]: This looks like lying to CMake because the COMMAND does not create any `final_${output}.tplg` file.

**Review by marc-hb — CHANGES_REQUESTED** [review]
Main issue is the "fake?" final_output.tplg file, other comments below.

File: tools/topology/topology1/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	sof/*.m4
+)
+
+add_custom_target(abi
+	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/get_abi.sh ${SOF_ROOT_SOURCE_DIRECTORY}
+	DEPENDS ${TPLG_DEPS}
+	WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+	VERBATIM
+	USES_TERMINAL
+)
```
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology1/CMakeLists.txt:17]: I cant do that. The outout ofthe get_abi.sh script is different for topology1 and topology2

File: tools/topology/topology1/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+	add_custom_command(
+		OUTPUT ${output}.tplg
+		COMMAND alsatplg \$\${VERBOSE:+-v 1} -c ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf -o ${output}.tplg
+		MAIN_DEPENDENCY ${output}.conf
+		USES_TERMINAL
+	)
+
+	add_custom_command(
+		OUTPUT final_${output}.tplg
+		COMMAND cp ${CMAKE_CURRENT_BINARY_DIR}/${output}.tplg ${CMAKE_CURRENT_BINARY_DIR}/..
```
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology1/CMakeLists.txt]: done. thanks!

File: tools/topology/topology1/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/topology/topology1/CMakeLists.txt:17]: Should this be renamed to `abi_v1` then? Just wondering. Sorry I forgot again what is the purpose of this, I mean which ABI it is. Add a comment?

File: tools/topology/topology1/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+# verbose or not.
+	add_custom_command(
+		OUTPUT ${output}.tplg
+		COMMAND alsatplg \$\${VERBOSE:+-v 1} -c ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf -o ${output}.tplg
+		MAIN_DEPENDENCY ${output}.conf
+		USES_TERMINAL
+	)
+
+	add_custom_target(topologies1_${output} DEPENDS ${output}.tplg)
+	add_dependencies(topologies1_output topologies1_${output})
```
Comment (marc-hb) [inline tools/topology/topology1/CMakeLists.txt:212]: The previous code used singular "topology" for a single file and plural "topologies" for the target that depends on them all. I think this made it very clear that the `add_dependencies(topologies topology_${i})` line is growing a list, even for people who don't know anything about CMake. Can you please preserve that?

File: tools/topology/topology1/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+	"sof-smart-amplifier-nocodec\;sof-smart-amplifier-nocodec"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015\;-DPLATFORM=jsl-rt1015"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015-xperi\;-DPLATFORM=jsl-rt1015\;-DINCLUDE_IIR_EQ=1"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-mx98360a\;-DPLATFORM=jsl-dedede"
+)
+
+add_custom_target(topologies1_output ALL)
+add_custom_target(topologies1 ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory . ${SOF_TOPOLOGY_BINARY_DIRECTORY}
+    DEPENDS topologies1_output
```
Comment (marc-hb) [inline tools/topology/topology1/CMakeLists.txt:155]: I find the names `topologies1_output` and `topologies1` not very clear. How about:
```
add_custom_target(topologies1 ALL)
add_custom_target(topologies1_copy ALL
   COMMAND copy_directory
```
End users should rarely ever invoke "make topologies1" directly. The very few who do should know what they're doing.

Or even simpler, have you tried to combine these two into a single `add_custom_target`? Shorter code and naming issue goes away

```
add_custom(topologies1 ALL
          COMMAND copy_directory...
...
)
...
add_dependencies(topologies1 topology1_${output})
```
The dependencies are obviously built before the COMMAND is run.

**Review by marc-hb — COMMENTED** [review]
Logic looks good. Sorry for nit-picking on the code style and symbol names but it's not like we have many full-time CMake experts in the project (none?) so reducing cognitive strain makes a significant difference.

**Comment by ranj063** [comment]
> Logic looks good. Sorry for nit-picking on the code style and symbol names but it's not like we have many full-time CMake experts in the project (none?) so reducing cognitive strain makes a significant difference.

fixed now. Thanks for your help!

File: tools/topology/topology1/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology1/CMakeLists.txt:17]: ther's only one ABI and its for the firmware. What should I say in the comment? that this is the firmware ABI?

[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/topology/topology1/CMakeLists.txt:17]: Two different formats for the same ABI, got it thanks for your patience.

**Review by marc-hb — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI showing unrelated failure.

File: tools/topology/topology1/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/topology/topology1/CMakeLists.txt]: Just learned the really hard way this is racing with the build of development topologies: #5067

**Pull request merged**